### PR TITLE
pgclusters.crunchydata.com CRD Cleanups + Additions

### PR DESCRIPTION
**Pull request by jkatz**
This patch series contains a few cleanups and changes to improve the overall usability of the `pgclusters.crunchydata.com` custom resources.

Changes include:

- The Service Port for monitoring replicas is now appropriately adjusted when monitoring is toggled.
- `ServiceType` (e.g. `ClusterIP`, `LoadBalancer`, `NodePort`) is now controlled with an attribute on `pgclusters.crunchydata.com` for the cluster-wide service, and the replica specific one on `pgreplicas.crunchydata.com`. The `ServiceType` can also be updated on existing clusters.
- pgBadger can now by toggled with an attribute on the `pgclusters.crunchydata.com` custom resource. This also adds the `--enable-pgbadger` and `--disable-pgbadger` flags on the `pgo update cluster` command.
- Removed the use of labels that had previous provided functionality but were superseded by CRD attributes, including labels for synchronous replication, pod anti-affinity and custom configuration.
- The `autofail` label is removed. The autofail/HA functionality of a cluster is now controlled by the `disableAutofail` attribute on `pgclusters.crunchydata.com`. A cluster with HA enabled sets `disableAutofail` to `false` (the default).
- The storage (or repo) type for pgBackRest is now controlled by a CRD attribute on `pgclusters.crunchydata.com`. In particularly, the attribute, `backrestStorageTypes` accepts an array of storage types that include:
    -  `posix` (the default)
    - `s3`
`local` is available for backwards compatibility purposes, but is effectively the same as `posix`. Combining `posix` and `s3` is the equivalent of sending backups and archives to both a `posix` and `s3` repository.
This eliminates the previous label that was used to store these attributes.
- For a multi-repository pgBackRest setup (`posix` + `s3`), the `pgo backup` command will now, by default, take backups simultaneously to both of those repositories.
- Allow for node affinity to be defined directly on the `pgclusters.crunchydata.com` and `pgreplicas.crunchydata.com` custom resource.
- Add `--node-affinity-type` flag on `pgo create cluster`, `pgo scale`, and `pgo restore`. Accepted values are `preferred` and `required` (though `prefer` and `require` are allowed).

### CODE REVIEW

File: cmd/pgo/cmd/update.go [hunk 1]
```diff
@@ -92,6 +96,8 @@ func init() {
 	UpdateClusterCmd.Flags().BoolVar(&DisableAutofailFlag, "disable-autofail", false, "Disables autofail capabitilies in the cluster.")
 	UpdateClusterCmd.Flags().BoolVar(&DisableMetrics, "disable-metrics", false,
 		"Disable the metrics collection sidecar. May cause brief downtime.")
+	UpdateClusterCmd.Flags().BoolVar(&DisablePGBadger, "disable-pgbadger", false,
+		"Disable the pgBAdger sidecar. May cause brief downtime.")
```
Comment (ValClarkson) [inline cmd/pgo/cmd/update.go:100]: pgBAdger  ->  pgBader

File: internal/operator/cluster/pgbadger.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		// if we fail to update the service, warn, but continue on
+		if _, err := clientset.CoreV1().Services(svc.Namespace).Update(ctx, svc, metav1.UpdateOptions{}); err != nil {
+			log.Warn(err)
+		}
+	}
+
+	return nil
+}
+
+// RemoveExporter disables the ability for a PostgreSQL cluster to run a
```
Comment (ValClarkson) [inline internal/operator/cluster/pgbadger.go:96]: RemovePGBadger

Comment (jkatz) [reply internal/operator/cluster/pgbadger.go]: Gee I wonder where I copy and pasted that from 🤔

File: internal/operator/cluster/service.go [hunk 3]
```diff
@@ -26,12 +26,22 @@ import (
...
 	"github.com/crunchydata/postgres-operator/internal/operator"
+	crv1 "github.com/crunchydata/postgres-operator/pkg/apis/crunchydata.com/v1"
 	log "github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
+// serviceInfo is a structured way of compiling all of te info required to
```
Comment (ValClarkson) [inline internal/operator/cluster/service.go:37]: typo in the comment
te info required -> the info required

**Review by ValClarkson — APPROVED** [review]

**Review by ValClarkson — APPROVED** [review]
lgtm

**Review by ValClarkson — APPROVED** [review]
lgtm

**Review by ValClarkson — APPROVED** [review]

**Review by ValClarkson — APPROVED** [review]

**Pull request merged**